frontends/web/src/utils/walletconnect-eth-sign-handlers.ts

```diff
@@ -0,0 +1,179 @@
+  const accountDetail = await getEthAccountCodeAndNameByAddress(address);
+  if (!accountDetail.success) {
+    console.log('Failed in fetching account name and code'); //silently fails
+    return { accountName: '', accountCode: '' };
+  }
+  const { code, name } = accountDetail;
+  return { accountName: name, accountCode: code };
+};
+
+export async function handleWcEthSignRequest(method: string, args: TEthSignHandlerParams) {
+  switch (method) {
+  case EIP155_SIGNING_METHODS.ETH_SIGN:
+  case EIP155_SIGNING_METHODS.PERSONAL_SIGN:
+    await ethSignHandler(args, method);
+    break;
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA:
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V3:
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V4:
+    await ethSignTypedDataHandler(args);
+    break;
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TRANSACTION:
+  case EIP155_SIGNING_METHODS.ETH_SEND_TRANSACTION:
+    await ethSignOrSendTransactionHandler(args, method);
+    break;
+  default:
+    console.log(`${method} is unsupported`);
+  }
+}
+
+/**
+     * Wallet Connect's ETH_SIGN gives the params as [address, message]
+     * while PERSONAL_SIGN gives them as [message, address]
+*/
+
+async function ethSignHandler({ params, launchSignDialog, topic, id, currentSession }: TEthSignHandlerParams, method: string) {
+  const isPersonalSign = method === EIP155_SIGNING_METHODS.PERSONAL_SIGN;
+  const requestParams = params.request.params;
+  const accountAddress = isPersonalSign ? requestParams[1] : requestParams[0];
+  const signingData = isPersonalSign ? requestParams[0] : requestParams[1];
+  const { accountName, accountCode } = await fetchAccountNameAndAddress(accountAddress);
+  const apiCaller = async () => {
+    const result = await ethSignMessage(accountCode, signingData);
+    if (!result.success) {
+      return { success: false, error: result };
+    }
+    return {
+      response: { id, jsonrpc: '2.0', result: result.signature },
+      success: true
+    };
+  };
+
+  launchSignDialog({
+    topic,
+    id,
+    apiCaller,
+    dialogContent: {
+      signingData,
+      currentSession,
+      accountName,
+      accountAddress,
+      chain: params.chainId,
+      method: t('walletConnect.signingRequest.method.signMessage')
+    }
+  });
+}
+
+async function ethSignTypedDataHandler({ params, launchSignDialog, topic, id, currentSession }: TEthSignHandlerParams) {
+  const requestParams = params.request.params;
+  const accountAddress = requestParams[0];
+  const data = requestParams[1];
+  const typedData = JSON.parse(data);
+  const { accountName, accountCode } = await fetchAccountNameAndAddress(accountAddress);
+  const apiCaller = async () => {
+    // If the typed data to be signed includes its own chainId, we use that, otherwise use the id in the params
+    const chainId = typedData.domain.chainId ?
+      +typedData.domain.chainId :
+      +params.chainId.replace(/^eip155:/, '');
```

Comment:
Is the  `+` to convert to Number? In other places in the codebase we use `Number('123')`, same in `ethSignOrSendTransactionHandler`

Reply:
that's probably from my code in the POC, I like the unary plus :)))

Reply:
please use `Number()` as we use in other places in the codebase, and is less mental overhead when reading the code. unary plus and Number apparently behaves the same.